plugins/org.eclipse.fordiac.ide.model.ui/src/org/eclipse/fordiac/ide/model/ui/editors/AbstractBreadCrumbEditor.java

```diff
@@ -141,16 +142,28 @@ protected Composite createPageContainer(final Composite parent) {
 		return pageContainer;
 	}
 
-	protected void handleBreadCrumbSelection(final Object element) {
-		if (element instanceof final SubApp subapp && subapp.isUnfolded()) {
-			breadcrumb.setInput(getFBNetworkContainer(subapp));
-			HandlerHelper.showExpandedSubapp(subapp, getActiveEditor());
+	protected void handleBreadCrumbSelection(final SelectionChangedEvent event) {
+		final Object element = ((StructuredSelection) event.getSelection()).getFirstElement();
+		if (element instanceof final SubApp subapp && subapp.isUnfolded()
+				&& !(event instanceof GoIntoSubappSelectionEvent)) {
+			showAndSelectExpandedSubapp(subapp);
 		} else {
-			final int pagenum = modelToEditorNum.computeIfAbsent(element, this::createEditor).intValue();
-			if (-1 != pagenum) {
-				setActivePage(pagenum);
-				getSite().getPage().getNavigationHistory().markLocation(getNavigationLocationProvider());
-			}
+			openEditor(element);
+		}
+	}
+
+	private void showAndSelectExpandedSubapp(final SubApp subapp) {
+		final EObject container = getFBNetworkContainer(subapp);
+		breadcrumb.setInput(container);
```

Comment:
isn't setInput resulting in the end such that openEditor is called and therefore you now are opening the editor twice?